x-pack/plugins/observability_ai_assistant/public/hooks/use_json_editor_model.ts

```diff
@@ -28,13 +25,20 @@ export const useJsonEditorModel = ({
 
   const [initialJsonValue, setInitialJsonValue] = useState<string | undefined>(initialJson);
 
+  const SCHEMA_URI = `http://elastic.co/${functionName}.json`;
+  const [modelUri, setModelUri] = useState<monaco.Uri | undefined>();
+
+  useEffect(() => {
+    setModelUri(Uri.parse(SCHEMA_URI));
+  }, [SCHEMA_URI, functionName]);
```

Comment:
`Uri` is part of the imports.

Reply:
Why would we want to run this once on mount? A user might select a different function which necessitates a recreation of the model.

Reply:
Right, that's fair. Yeah, my question is not about `Uri` but about `functionName`. Maybe it understands that if functionName changes, so does `SCHEMA_URI`, but I'm skeptical. (Asking because I want to make sure that the lint rules actually work).

Reply:
Just verified, the lint rule does work: 

<img width="686" alt="Screenshot 2024-01-08 at 12 24 03" src="https://github.com/elastic/kibana/assets/535564/64604409-5290-442a-b83d-169b06fbcb6a">

I can remove `functionName` though.